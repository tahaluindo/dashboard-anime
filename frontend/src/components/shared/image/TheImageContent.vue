<template>
  <img loading="lazy" class="w-full object-cover" :src="src" :alt="description" />
</template>

<script lang="ts">
import { defineComponent } from "vue";

export default defineComponent({
  props: {
    src: String,
    description: String,
  },
});
</script>