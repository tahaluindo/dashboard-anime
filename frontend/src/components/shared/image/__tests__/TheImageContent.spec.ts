import { mount } from "@vue/test-utils";
import TheImageContent from "../TheImageContent.vue";

describe("TheImageContent test suit", () => {
  it("renders the component with the props passed", () => {
    const wrapper = mount(TheImageContent, {
      props: {
        src: "https:/localhost/test",
        description: "image",
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
