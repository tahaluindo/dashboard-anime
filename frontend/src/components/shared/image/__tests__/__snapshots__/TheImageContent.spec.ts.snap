// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TheImageContent test suit renders the component with the props passed 1`] = `"<img loading=\\"lazy\\" class=\\"w-full object-cover\\" src=\\"https:/localhost/test\\" alt=\\"image\\">"`;
